pkg/kubelet/logs/container_log_manager.go

```diff
@@ -0,0 +1,400 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/wait"
+	internalapi "k8s.io/kubernetes/pkg/kubelet/apis/cri"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+const (
+	// logMonitorPeriod is the period container log manager monitors
+	// container logs and performs log rotation.
+	logMonitorPeriod = 10 * time.Second
```

Comment:
is listing all containers every 10 seconds necessary?  Would it be better to use information from a cache?

Reply:
To use cache, we need to add a List function in cache.
I'm fine with that actually.

Reply:
I can do that as a following up PR. It does need some more change. Will file an issue and add TODO here.

Reply:
sgtm

Reply:
Done https://github.com/kubernetes/kubernetes/issues/59998